cmd/main.go

```diff
@@ -181,10 +186,33 @@ func (f *Aggregator) Aggregate(targets []string, output io.Writer) {
 					}
 					if existingMf, ok := allFamilies[mfName]; ok {
 						for _, m := range mf.Metric {
-							existingMf.Metric = append(existingMf.Metric, m)
+							for _, r := range m.Label {
+								match, _ := regexp.MatchString(*targetLabelRegexp, r.GetValue())
+								if match {
+									existingMf.Metric = append(existingMf.Metric, m)
+								}
+							}
 						}
 					} else {
-						allFamilies[*mf.Name] = mf
+						im := 1
+						for _, m := range mf.Metric {
+							for _, r := range m.Label {
+								match, _ := regexp.MatchString(*targetLabelRegexp, r.GetValue())
+								if match {
+									newFamilies = mf
+									if im == 1 {
+										newFamilies.Metric = newFamilies.Metric[:1]
+										newFamilies.Metric[0] = m
```

Comment:
This looks a bit funny to me. So it's going to copy `mf` to `newFamilies`  then basically remove all except the first element and replace it with the current match. Why not just assign a new slice containing only `m`?

Reply:
Huh, maybe you can propose some solution how to do it? I'm not developer, and this part little bit hard for me :)

Reply:
I think I found solution, https://github.com/warmans/prometheus-aggregate-exporter/pull/11/commits/1a05eed714407604193988677984cfb81ea2864c